decentraland/unity-renderer
# refactor: extendible settings creation

## What does this PR change?

Makes `Settings` creation extendible, so it can be overrided in desktop.
For example: change the path where the graphic quality settings preset are, or change default general settings.

## How to test the changes?

1. Go to: https://play.decentraland.zone/?renderer-branch=feat/desktop-default-settings
2. Open settings menu
3. Click on `Reset All` button
4. All settings should be set to default as expected

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
